tests/aws/services/events/test_archive_and_replay.py

```diff
@@ -0,0 +1,971 @@
+            event_source_arn = response["EventBusArn"]
+
+        # setup rule
+        rule_name = f"test-rule-{short_uid()}"
+        response = events_put_rule(
+            Name=rule_name,
+            EventBusName=event_bus_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+        )
+        rule_arn = response["RuleArn"]
+
+        # setup sqs target
+        queue_url, queue_arn = create_sqs_events_target()
+        target_id = f"target-{short_uid()}"
+        aws_client.events.put_targets(
+            Rule=rule_name,
+            EventBusName=event_bus_name,
+            Targets=[
+                {"Id": target_id, "Arn": queue_arn},
+            ],
+        )
+
+        # put events to archive
+        num_events = 3
+        entries = []
+        for _ in range(num_events):
+            entry = {
+                "Source": TEST_EVENT_PATTERN["source"][0],
+                "DetailType": TEST_EVENT_PATTERN["detail-type"][0],
+                "Detail": json.dumps(EVENT_DETAIL),
+                "EventBusName": event_bus_name,
+            }
+            entries.append(entry)
+
+        archive_name = f"test-archive-{short_uid()}"
+        archive_arn = put_event_to_archive(
+            archive_name,
+            TEST_EVENT_PATTERN,
+            event_bus_name,
+            event_source_arn,
+            entries,
+        )["ArchiveArn"]
+        sqs_collect_messages(
+            aws_client, queue_url, expected_events_count=num_events, wait_time=5, retries=12
+        )  # reset queue for replay
+
+        # start replay
+        replay_name = f"test-replay-{short_uid()}"
+        response_start_replay = aws_client.events.start_replay(
+            ReplayName=replay_name,
+            Description="description of the replay",
+            EventSourceArn=archive_arn,
+            EventStartTime=event_start_time,
+            EventEndTime=event_end_time,
+            Destination={
+                "Arn": event_source_arn,
+                "FilterArns": [
+                    rule_arn,
+                ],
+            },
+        )
+
+        snapshot.add_transformer(
+            [
+                snapshot.transform.regex(event_bus_name, "<event-bus-name>"),
+                snapshot.transform.regex(rule_name, "<rule-name>"),
+                snapshot.transform.regex(archive_name, "<archive-name>"),
+                snapshot.transform.regex(replay_name, "<replay-name>"),
+                snapshot.transform.key_value("ReceiptHandle", reference_replacement=False),
+                snapshot.transform.key_value("MD5OfBody", reference_replacement=False),
+                snapshot.transform.key_value("ReplayName", reference_replacement=False),
+            ]
+        )
+        snapshot.match("start-replay", response_start_replay)
+
+        # replaying an archive mostly takes at least 5 minutes on AWS
+        wait_for_replay_in_state(aws_client, replay_name, "COMPLETED", retries=35, sleep=10)
```

Comment:
Nicely waiting for separate events separately following R7: https://github.com/localstack/localstack/blob/master/docs/testing/README.md#r07
👏